metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricQueryBuilder.java

```diff
@@ -261,18 +299,18 @@ static Expression createStringLiteralExpression(String value) {
   }
 
   String getTimestampAttributeId(String tenantId, String attributeScope) {
-    Iterator<AttributeMetadata> attributeMetadataIterator =
-        attributesServiceClient.findAttributes(
-            Map.of(MetricAnomalyDetector.TENANT_ID_KEY, tenantId),
-            AttributeMetadataFilter.newBuilder().addScopeString(attributeScope).build());
-
-    while (attributeMetadataIterator.hasNext()) {
-      AttributeMetadata metadata = attributeMetadataIterator.next();
-      if (metadata.getKey().equals(START_TIME_ATTRIBUTE_KEY)) {
-        return metadata.getId();
-      }
+    try {
+      return attributeServiceCache.get(List.of(tenantId, attributeScope));
```

Comment:
Please add a test verifying that the first call hits the mock and second call is served by the cache